code/modules/assembly/signaler.dm

```diff
@@ -241,6 +241,15 @@
 	icon_state = "vortex core"
 	anomaly_type = /obj/effect/anomaly/bhole
 
+/obj/item/assembly/signaler/anomaly/activate()
+	. = ..()
+	var/obj/effect/anomaly/anomaly_path = anomaly_type
```

Comment:
new anomaly_type() works fine, you only need to define a typepath by type like this if you are using initial(), since anomaly_type isn't actually a /obj/effect/anomaly its a typepath.

Reply:
Imma keep it real, I only know like half of those words.